Entity reference fields non autocomplete

steveworley opened the pull request:
- Updates `AudtiResponse.php` so remediation advice can use tokens
- Adds `EntityReferenceAutocomplete.php`

Recently I ran into an issue with a site that would cripple the server, basically when the entity reference field was loaded as a select list with > 100 references we see exponential degradation. It has a number of calls to render which can then expand into thousands of other calls based on the entities render function.

Some anecdotal stats:
- Referenced entities: 4796
- Time-to-first-byte 5.6mins

An interesting thing with this issue is that `php_max_execution` for the script was never reached so it would block a process for the entire duration. So I think that it is a valuable check to add for govcms checks given the shared resourcing.

The check will attempt to find entity reference fields that aren't configured as autocomplete fields and see if they're referencing content that has more than a threshold which we can control via the yml configs.

Commit:
Add entity reference checks

Commit:
Add entity reference field check to govcms_saas

Commit:
Change remediation reporting

- Updates remdiation reporting
- Fixes issue with remediation not using tokens

Commit:
Remove erroneous whitespace

seanhamlin commented on src/AuditResponse/AuditResponse.php:
```diff
@@ -79,7 +79,8 @@ public function getDescription() {
   public function getRemediation() {
-    return $this->check->getInfo()->remediation;
+    $tokens = $this->check->getTokens();
+    return strtr($this->check->getInfo()->remediation, $tokens);
```
This is a great change.

seanhamlin commented on src/Check/D7/EntityReferenceAutocomplete.php:
```diff
@@ -0,0 +1,176 @@
+    $valid = 0;
+
+    try {
+      $output = $this->context->drush->sqlQuery('SELECT fc.field_name, fc.data, fci.data FROM {field_config} fc JOIN {field_config_instance} fci ON fc.id = fci.field_id WHERE fc.type = \'entityreference\'');
```
Can we do double quotes on the outside, and unescaped single on the inside. I hope all checks follow this standard (the quoting escaping is very complex).

seanhamlin commented on src/Check/D7/EntityReferenceAutocomplete.php:
```diff
@@ -0,0 +1,176 @@
+    try {
+      $output = $this->context->drush->sqlQuery('SELECT fc.field_name, fc.data, fci.data FROM {field_config} fc JOIN {field_config_instance} fci ON fc.id = fci.field_id WHERE fc.type = \'entityreference\'');
+    } catch (\Exception $e) {
+      return AuditResponse::AUDIT_FAILURE;
```
You can throw an exception if this genuinely should never happen. Failure is more for when the check has failed to pass and there is some action to fix it needed.

seanhamlin commented on src/Check/D7/EntityReferenceAutocomplete.php:
```diff
@@ -0,0 +1,176 @@
+    $node_types = "'" . implode("','", $node_types) . "'";
+
+    try {
+      $output = $this->context->drush->sqlQuery('SELECT count(node.nid) as count FROM {node} node WHERE node.type in (' . $node_types . ')');
```
double on the outside, single on the inside again.

seanhamlin reviewed (changes requested):
Let's chat tomorrow. Love this PR.

Commit:
Code review changes

- Updates querie to use double quotes
- Throws an exception if the initial query fails
